src/System.Private.ServiceModel/src/System/IdentityModel/CryptoHelper.cs

```diff
@@ -61,7 +97,305 @@ internal static bool IsAsymmetricAlgorithm(string algorithm)
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return SHA1.Create();
+#pragma warning restore CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                case SHA256String:
+                case SecurityAlgorithms.Sha256Digest:
+                    return SHA256.Create();
+                default:
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new InvalidOperationException(SR.Format(SR.UnsupportedCryptoAlgorithm, algorithm)));
+            }
+        }
+
+        private static object GetDefaultAlgorithm(string algorithm)
+        {
+            if (string.IsNullOrEmpty(algorithm))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException("algorithm"));
+            }
+
+            switch (algorithm)
+            {
+                //case SecurityAlgorithms.RsaSha1Signature:
+                //case SecurityAlgorithms.DsaSha1Signature:
+                // For these algorithms above, crypto config returns internal objects.
+                // As we cannot create those internal objects, we are returning null.
+                // If no custom algorithm is plugged-in, at least these two algorithms
+                // will be inside the delegate dictionary.
+                case SecurityAlgorithms.Sha1Digest:
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return SHA1.Create();
+#pragma warning restore CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                case SecurityAlgorithms.ExclusiveC14n:
+                    throw ExceptionHelper.PlatformNotSupported();
+                case SHA256String:
+                case SecurityAlgorithms.Sha256Digest:
+                    return SHA256.Create();
+                case SecurityAlgorithms.Sha512Digest:
+                    return SHA512.Create();
+                case SecurityAlgorithms.Aes128Encryption:
+                case SecurityAlgorithms.Aes192Encryption:
+                case SecurityAlgorithms.Aes256Encryption:
+                case SecurityAlgorithms.Aes128KeyWrap:
+                case SecurityAlgorithms.Aes192KeyWrap:
+                case SecurityAlgorithms.Aes256KeyWrap:
+                    return Rijndael.Create();
+                case SecurityAlgorithms.TripleDesEncryption:
+                case SecurityAlgorithms.TripleDesKeyWrap:
+#pragma warning disable CA5350 // Do Not Use Weak Cryptographic Algorithms
+                    return TripleDES.Create();
+#pragma warning restore CA5350 // Do Not Use Weak Cryptographic Algorithms
+                case SecurityAlgorithms.HmacSha1Signature:
+                    byte[] key = new byte[64];
+                    new RNGCryptoServiceProvider().GetBytes(key);
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return new HMACSHA1(key);
```

Comment:
Seems weird that for HMACSHA1 you're doing manual-random, but HMACSHA256 you let the ctor do it.  Why not just let the HMACSHA1 ctor do it?

Reply:
Original code used the parameter which specifies whether to use FIPS or not, and there wasn't an overload where you specify that without specifying FIPS. I missed that you just let it do it's own thing in the default constructor. I'll fix.

---
In reply to: [274001237](https://github.com/dotnet/wcf/pull/3538#discussion_r274001237) [](ancestors = 274001237)